### ISSUE TITLE
NamedArguments rule: false positive with trailing lambda

### ISSUE THREAD

**Issue by wilerson** [detekt/detekt#3658]
```kotlin
fun foo(a: Int, b: Int, c: Int, block: ((Int) -> Int)) {}

fun test() {
    foo(a = 1, b = 2, c = 3) { it }
}
```

**Comment by dimsuz** [comment]
I have this failing with suspend functions:

```kotlin
      combine(
        mapRotationChanges,
        overlayMediator.controlVisibility(Controls.Compass),
        controlsVisibilityProvider.visibility,
        ::mergeCompassControl
      )
```

where in this case `combine` is defined as:

```kotlin
fun <T1, T2, T3, R> combine(
    flow: Flow<T1>,
    flow2: Flow<T2>,
    flow3: Flow<T3>,
    @BuilderInference transform: suspend (T1, T2, T3) -> R
): Flow<R>
```

Also a separate question: I wonder is there a way to disable this rule for some functions specifically. For example kotlin coroutines library has 8 overloads of `combine` for different flow count and specifying `flow`, `flow2`,... each time would be strange. On the other hand I don't want to increase threshold for generic cases.
